pkg/scheduler/framework/types.go

```diff
@@ -467,18 +467,10 @@ func NewNodeInfo(pods ...*v1.Pod) *NodeInfo {
 	return ni
 }
 
-// Node returns overall information about this node.
-func (n *NodeInfo) Node() *v1.Node {
-	if n == nil {
```

Comment:
you probably already thought of this, but do we know if we call `Node()` on a nil struct anywhere? if we do, such a call would safely return nil before, but with this change we'd get a nil deref panic

Reply:
yeah, this is the most effort for this PR, having to check all the callers and find out where do they get the nodeInfo struct. Most of them are from snapshot and few from construction. So I think it should fine.